README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->
<!-- badges: start -->

[![Lifecycle:
experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://www.tidyverse.org/lifecycle/#experimental)
<!-- badges: end -->

# ACDIVOCAthemes

ACDIVOCAthemes packages contains color palette and `ggplot2` theme based
on international development NGO ACDI/VOCA’s colors.

## Installation

You can install the released version of AVthemes from Github with:

``` r
# install.packages("devtools")
devtools::install_github("ACDIVOCATech/ACDIVOCAthemes")
```

## Example

``` r
library(extrafont)
library(ggplot2)
library(dplyr)
library(ACDIVOCAthemes)
```

### `theme_AV()`

``` r
## basic example code
mpg %>% 
  ggplot(aes(displ)) +
  geom_histogram(aes(fill = rev(class)), col = "black", size = 0.1) +
  scale_x_continuous(expand = c(0.01, 0)) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_brewer(name = "CUIDs") +
  labs(title = "ACDI/VOCA themes",
       subtitle = "Visualizations are key!",
       x = "$ spent per Pound of Fertilizer", y = "Number",
       caption = "Made by Data Science Team") +
  theme_AV(ticks = TRUE)
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

<img src="man/figures/README-example-1.png" width="100%" />

### `theme_AVgray`

``` r
mpg %>% 
  ggplot(aes(displ)) +
  geom_histogram(aes(fill = rev(class)), col = "black", size = 0.1) +
  scale_x_continuous(expand = c(0.01, 0)) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_brewer(name = "CUIDs") +
  labs(title = "ACDI/VOCA themes",
       subtitle = "Visualizations are key!",
       x = "$ spent per Pound of Fertilizer", y = "Number",
       caption = "Made by Data Science Team") +
  theme_AVgray(ticks = TRUE)
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

<img src="man/figures/README-unnamed-chunk-3-1.png" width="100%" />


NEWS.md
# ACDIVOCAthemes 0.1.1

* Remove "Calibri" as default font & remove all font options from theme functions.
  * See [issue #2](https://github.com/ACDIVOCATech/ACDIVOCAthemes/issues/2)

# ACDIVOCAthemes 0.1.0

* Added a `NEWS.md` file to track changes to the package.
* Create package.
* Edit README & Add DESCRIPTION.
* Add tests.


tests/testthat.R
library(testthat)
library(AVthemes)

test_check("AVthemes")


tests/testthat/test-themes.R
context("test-themes")

# Theme AV ----
test_that("theme_AV works", {
  thm <- theme_AV()
  expect_s3_class(thm, "theme")
  ## font
  expect_equal(thm$text$family, "Calibri")
  expect_equal(thm$plot.title$family, "Calibri")
  expect_equal(thm$legend.title$family, "Calibri")
  expect_equal(thm$legend.text$family, "Calibri")
  ## size
  expect_equal(thm$text$size, 14)
  expect_equal(thm$plot.title$size, 18)
  expect_equal(thm$plot.subtitle$size, 12)
  expect_equal(thm$axis.text$size, 12)
  expect_equal(thm$axis.title$size, 14)
  expect_equal(thm$legend.text$size, 9)
  expect_equal(thm$legend.title$size, 10)
  ## color
  expect_equal(thm$text$colour, "#F9FEFF")
  expect_equal(thm$plot.title$colour, "#F9FEFF")
  expect_equal(thm$plot.subtitle$colour, "#F9FEFF")
  expect_equal(thm$axis.text$colour, "#F9FEFF")
  expect_equal(thm$axis.title$colour, "#F9FEFF")
  expect_equal(thm$legend.text$colour, "#F9FEFF")
  expect_equal(thm$legend.title$colour, "#F9FEFF")
  expect_equal(thm$legend.position, "bottom")
  ## background
  expect_equal(thm$plot.background$fill, "#009BA7")
  ## panel
  expect_equal(thm$panel.grid$colour, "#505159")
})

# Theme AVgrey ----
test_that("theme_AVgrey works", {
  thm <- theme_AVgrey()
  expect_s3_class(thm, "theme")
  ## font
  expect_equal(thm$text$family, "")
  expect_null(thm$plot.title$family)
  expect_null(thm$legend.title$family)
  expect_null(thm$legend.text$family)
  ## size
  expect_equal(thm$text$size, 14)
  expect_equal(thm$plot.title$size, 18)
  expect_equal(thm$plot.subtitle$size, 12)
  expect_equal(thm$axis.text$size, 10)
  expect_equal(thm$axis.title$size, 14)
  expect_equal(thm$legend.text$size, 9)
  expect_equal(thm$legend.title$size, 10)
  ## color
  expect_equal(thm$text$colour, "#fee8c8")
  expect_equal(thm$plot.title$colour, "#FFD235")
  expect_equal(thm$plot.subtitle$colour, "#fee8c8")
  expect_equal(thm$axis.text$colour, "#fee8c8")
  expect_equal(thm$axis.title$colour, "#fee8c8")
  expect_equal(thm$legend.text$colour, "#ffffff")
  expect_equal(thm$legend.title$colour, "#ffffff")
  expect_equal(thm$legend.position, "bottom")
})


tests/testthat/test-palettes.R
context("test-palettes")

test_that("multiplication works", {
  expect_equal(2 * 2, 4)
})


R/add_logo.R
## add_logo function from Thomas Mock
## check out the explanation in his blog post: https://themockup.netlify.com/posts/2019-01-09-add-a-logo-to-your-plot/


#' @title add_logo
#' @description add ACDI/VOCA logo
#' @param plot_path PARAM_DESCRIPTION
#' @param logo_position PARAM_DESCRIPTION
#' @param logo_scale PARAM_DESCRIPTION, Default: 10
#' @return OUTPUT_DESCRIPTION
#' @rdname add_logo
#' @export
#' @importFrom magick image_read image_info image_scale image_composite

add_logo <- function(plot_path, logo_position, logo_scale = 10){

  ## Requires magick R Package https://github.com/ropensci/magick

  ## Useful error message for logo position
  if (!logo_position %in% c("top right", "top left", "bottom right", "bottom left")) {
    stop("Error Message: Uh oh! Logo Position not recognized\n  Try: logo_positon = 'top left', 'top right', 'bottom left', or 'bottom right'")
  }

  ## logo path
  logo_path <- system.file("logo/logo_simple.png", package = "AVthemes")

  ## read in raw images
  plot <- magick::image_read(plot_path)
  logo_raw <- magick::image_read(logo_path)

  ## get dimensions of plot for scaling
  plot_height <- magick::image_info(plot)$height
  plot_width <- magick::image_info(plot)$width

  ## default scale to 1/10th width of plot
  ## Can change with logo_scale
  logo <- magick::image_scale(logo_raw, as.character(plot_width/logo_scale))

  # Get width of logo
  logo_width <- magick::image_info(logo)$width
  logo_height <- magick::image_info(logo)$height

  ## Set position of logo
  ## Position starts at 0,0 at top left
  ## Using 0.01 for 1% - aesthetic padding

  if (logo_position == "top right") {
    x_pos = plot_width - logo_width - 0.01 * plot_width
    y_pos = 0.01 * plot_height
  } else if (logo_position == "top left") {
    x_pos = 0.01 * plot_width
    y_pos = 0.01 * plot_height
  } else if (logo_position == "bottom right") {
    x_pos = plot_width - logo_width - 0.01 * plot_width
    y_pos = plot_height - logo_height - 0.01 * plot_height
  } else if (logo_position == "bottom left") {
    x_pos = 0.01 * plot_width
    y_pos = plot_height - logo_height - 0.01 * plot_height
  }

  # Compose the actual overlay
  magick::image_composite(plot, logo, offset = paste0("+", x_pos, "+", y_pos))

}


R/AV_palette.R
## AV palettes

AV_palette <- list(
  ## AV Regular
  Regular = c(
    "#F6E0D2",  ## white paper
    "#DFA398",  ##
    "#9C6755",  ##
    "#659794",  ## AV turqoise
    "#EA967C",  ## AV lightorange
    "#F5C98E",  ## AV beige
    "#D65B5A",  ##
    "#586085"   ##
  ),
  ## AV Grey
  Grey = c(
    "#EFCBCB",  ##
    "#B47880",  ##
    "#824B51",  ##
    "#635761",  ##
    "#AD616C",  ##
    "#D1A391",  ##
    "#7D3450",  ##
    "#2B2D42"   ##
  ),
  Sequential_Blues = c(
    "#f7fbff",  ##
    "#deebf7",  ##
    "#c6dbef",  ##
    "#9ecae1",  ##
    "#6baed6",  ##
    "#4292c6",  ##
    "#2171b5",  ##
    "#08519c",  ##
    "#08306b"   ##
  ),
  Sequential_Grey = c(
    "#EFCBCB",  ##
    "#B47880",  ##
    "#824B51",  ##
    "#635761",  ##
    "#AD616C",  ##
    "#D1A391",  ##
    "#7D3450",  ##
    "#2B2D42"   ##
  )
)

#' @title AV palette
#' @description AV palette
#' @details Color set from Matt Shanks & `@ChevyRay``
#' @inheritDotParams ggplot2::discrete_scale
#' @param palette name of palette, Default: "Day"
#' @param n number of colors
#' @param type discrete or continuous
#' @param reverse reverse order, Default: FALSE
#' @seealso
#'  \code{\link[scales]{manual_pal}}
#'  [ggplot2::scale_color_discrete]
#'  [ggplot2::scale_fill_discrete]
#' @rdname AV_pal
#' @export
#' @importFrom scales manual_pal

AV_pal <- function(palette, n, type = c("discrete", "continuous"),
                      reverse = FALSE){
  AV <- AV_palette[[palette]]

  if (reverse == TRUE) {
    AV <- rev(AV)
  }

  if (missing(n)) {
    n <- length(AV)
  }

  type <- match.arg(type)

  if (type == "discrete" && n > length(AV)) {
    stop(glue::glue("Palette does not have {n} colors, maximum is {length(AV)}!"))
  }

  AV <- switch(type,
                  continuous = grDevices::colorRampPalette(AV)(n),
                  discrete = AV[1:n])

  AV <- scales::manual_pal(AV)

  return(AV)
}

#' @title scale_color_AV
#' @rdname AV_pal
#' @export
#' @importFrom ggplot2 discrete_scale

scale_color_AV <- function(palette = "Day", n, type = "discrete",
                              reverse = FALSE, ...){

  if (type == "discrete") {
    ggplot2::discrete_scale("color", "AV",
                            AV_pal(palette = palette, n = n, type = type,
                                      reverse = reverse), ...)
  } else { ## needs work...
    ggplot2::scale_color_gradientn(colors = AV_pal(palette = palette, n = n, type = type,
                                                      reverse = reverse)(8))
  }
}

#' @title scale_colour_AV
#' @rdname AV_pal
#' @export
#' @importFrom ggplot2 discrete_scale

scale_colour_AV <- scale_color_AV

#' @title scale_fill_AV
#' @rdname AV_pal
#' @export
#' @importFrom ggplot2 discrete_scale

scale_fill_AV <- function(palette = "Day", n, type = "discrete",
                             reverse = FALSE, ...){
  if (type == "discrete") {
    ggplot2::discrete_scale("fill", "AV",
                            AV_pal(palette = palette, n = n, type = type,
                                      reverse = reverse), ...)
  } else { ## needs work...
    ggplot2::scale_fill_gradientn(colors = AV_pal(palette = palette, n = n, type = type,
                                                     reverse = reverse)(8))
  }
}


R/theme_AV.R
#' @title theme_AV
#' @description ACDI/VOCA plotting theme
#' @param title.size PARAM_DESCRIPTION, Default: 18
#' @param text.size PARAM_DESCRIPTION, Default: 14
#' @param subtitle.size PARAM_DESCRIPTION, Default: 12
#' @param axis.title.size PARAM_DESCRIPTION, Default: 14
#' @param axis.text.size PARAM_DESCRIPTION, Default: 12
#' @param legend.title.size PARAM_DESCRIPTION, Default: 10
#' @param legend.text.size PARAM_DESCRIPTION, Default: 9
#' @param title.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param subtitle.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param text.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param axis.title.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param axis.text.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param legend.title.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param legend.text.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param legend.position PARAM_DESCRIPTION, Default: 'bottom'
#' @param ticks PARAM_DESCRIPTION, Default: FALSE
#' @return OUTPUT_DESCRIPTION
#' @rdname theme_AV
#' @export
#' @importFrom ggplot2 element_text
#' @importFrom grid unit

theme_AV <- function(
  title.size = 18,
  text.size = 14,
  subtitle.size = 12,
  axis.title.size = 14,
  axis.text.size = 12,
  legend.title.size = 10,
  legend.text.size = 9,
  title.color = "#F9FEFF",
  subtitle.color = "#F9FEFF",
  text.color = "#F9FEFF",
  axis.title.color = "#F9FEFF",
  axis.text.color = "#F9FEFF",
  legend.title.color = "#F9FEFF",
  legend.text.color = "#F9FEFF",
  legend.position = "bottom",
  ticks = FALSE
) {

  ## base on theme_minimal
  themeAV <- theme_minimal() +
    ## main theme
    theme(text = ggplot2::element_text(size = text.size, color = text.color),
          plot.title = element_text(face = "bold", size = title.size, color = title.color),
          plot.subtitle = element_text( size = subtitle.size, color = subtitle.color),
          # background/panel options
          plot.background = element_rect(color = NA, fill = "#009BA7"),
          panel.background = element_rect(color = NA),
          # panel options
          panel.grid = element_line(color = "#505159"),
          # axis options
          axis.text = element_text(size = axis.text.size, color = axis.text.color),
          axis.title = element_text(size = axis.title.size, color = axis.title.color),
          # legend options
          legend.text = element_text(size = legend.text.size, color = legend.text.color),
          legend.title = element_text(size = legend.title.size, color = legend.title.color),
          legend.position = legend.position,
          legend.key = element_rect(colour = "#505159", linetype = "solid"),
          legend.background = element_rect(color = "#505159", fill = "#505159", linetype = "solid")  ## #64719E  #8a2be2
    )

  if (ticks == FALSE) {
    themeAV <- themeAV + theme(axis.ticks = element_blank(),
                           axis.ticks.x = element_blank(),
                           axis.ticks.y = element_blank())
  } else {
    themeAV <- themeAV + theme(axis.ticks = element_line(size = 0.15, color = "#505159"),
                           axis.ticks.x = element_line(size = 0.15, color = "#505159"),
                           axis.ticks.y = element_line(size = 0.15, color = "#505159"),
                           axis.ticks.length = grid::unit(4, "pt"))
  }

  return(themeAV)

  ## add ACDIV/VOCA logo
}


#' @title FUNCTION_TITLE
#' @description FUNCTION_DESCRIPTION
#' @param title.size PARAM_DESCRIPTION, Default: 18
#' @param text.size PARAM_DESCRIPTION, Default: 14
#' @param subtitle.size PARAM_DESCRIPTION, Default: 12
#' @param axis.title.size PARAM_DESCRIPTION, Default: 14
#' @param axis.text.size PARAM_DESCRIPTION, Default: 12
#' @param legend.title.size PARAM_DESCRIPTION, Default: 10
#' @param legend.text.size PARAM_DESCRIPTION, Default: 9
#' @param title.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param subtitle.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param text.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param axis.title.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param axis.text.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param legend.title.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param legend.text.color PARAM_DESCRIPTION, Default: '#F9FEFF'
#' @param legend.position PARAM_DESCRIPTION, Default: 'bottom'
#' @param ticks PARAM_DESCRIPTION, Default: FALSE
#' @return OUTPUT_DESCRIPTION
#' @details DETAILS
#' @examples
#' \dontrun{
#' if(interactive()){
#'  #EXAMPLE1
#'  }
#' }
#' @seealso
#'  \code{\link[ggplot2]{margin}}
#'  \code{\link[grid]{unit}}
#' @rdname theme_AVgray
#' @export
#' @importFrom ggplot2 element_text
#' @importFrom grid unit

theme_AVgray <- function(
  title.size = 18,
  text.size = 14,
  subtitle.size = 12,
  axis.title.size = 14,
  axis.text.size = 12,
  legend.title.size = 10,
  legend.text.size = 9,
  title.color = "#F9FEFF",
  subtitle.color = "#F9FEFF",
  text.color = "#F9FEFF",
  axis.title.color = "#F9FEFF",
  axis.text.color = "#F9FEFF",
  legend.title.color = "#F9FEFF",
  legend.text.color = "#F9FEFF",
  legend.position = "bottom",
  ticks = FALSE
) {

  ## base on theme_minimal
  theme_AV_gray <- theme_gray() +
    ## main theme
    theme(text = ggplot2::element_text(size = text.size, color = text.color),
          plot.title = element_text(face = "bold", size = title.size, color = title.color),
          plot.subtitle = element_text(size = subtitle.size, color = subtitle.color),
          # background/panel options
          plot.background = element_rect(color = NA, fill = "#505159"),
          panel.background = element_rect(color = NA),
          # panel options
          panel.grid = element_line(color = "black"),
          # axis options
          axis.text = element_text(size = axis.text.size, color = axis.text.color),
          axis.title = element_text(size = axis.title.size, color = axis.title.color),
          # legend options
          legend.text = element_text(size = legend.text.size, color = legend.text.color),
          legend.title = element_text(size = legend.title.size, color = legend.title.color),
          legend.position = legend.position,
          legend.key = element_rect(colour = "black", linetype = "solid"),
          legend.background = element_rect(color = "#505159", fill = "#505159", linetype = "solid")  ## #64719E  #8a2be2
    )

  if (ticks == FALSE) {
    theme_AV_gray <- theme_AV_gray + theme(axis.ticks = element_blank(),
                               axis.ticks.x = element_blank(),
                               axis.ticks.y = element_blank())
  } else {
    theme_AV_gray <- theme_AV_gray + theme(axis.ticks = element_line(size = 0.15, color = "black"),
                               axis.ticks.x = element_line(size = 0.15, color = "black"),
                               axis.ticks.y = element_line(size = 0.15, color = "black"),
                               axis.ticks.length = grid::unit(4, "pt"))
  }

  return(theme_AV_gray)

  ## add ACDIV/VOCA logo
}
